src/Files.App/Views/NavigationInteractionTracker.cs

```diff
@@ -0,0 +1,205 @@
+﻿using Microsoft.UI.Composition;
+using Microsoft.UI.Composition.Interactions;
+using Microsoft.UI.Input;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Hosting;
+using Microsoft.UI.Xaml.Input;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Files.App.Views
+{
+	internal class NavigationInteractionTracker
+	{
+		public bool CanGoForward
+		{
+			get
+			{
+				_props.TryGetBoolean("CanGoForward", out bool val);
+				return val;
+			}
+			set
+			{
+				_props.InsertBoolean("CanGoForward", value);
+			}
+		}
+
+		public bool CanGoBack
+		{
+			get
+			{
+				_props.TryGetBoolean("CanGoBack", out bool val);
+				return val;
+			}
+			set
+			{
+				_props.InsertBoolean("CanGoBack", value);
+			}
+		}
+
+		private UIElement _rootElement;
+		private UIElement _backIcon;
+		private UIElement _forwardIcon;
+
+		private Visual _rootVisual;
+		private Visual _backVisual;
+		private Visual _forwardVisual;
+
+		private InteractionTracker _tracker = null!;
+		private VisualInteractionSource _source = null!;
+		private InteractionTrackerOwner _trackerOwner = null!;
+		private CompositionPropertySet _props = null!;
```

Comment:
```suggestion
		private InteractionTracker _tracker;
		private VisualInteractionSource _source;
		private InteractionTrackerOwner _trackerOwner;
		private CompositionPropertySet _props;
```

You don't need to declare them as `not null` since you assign them in the `constructor`
You may also consider to mark them as `readonly`